code/modules/organs/blood.dm

```diff
@@ -210,10 +219,16 @@ var/const/CE_STABLE_THRESHOLD = 0.5
 	if(!amt)
 		return 0
 
-	if(amt > vessel.get_reagent_amount("blood"))
-		amt = vessel.get_reagent_amount("blood") - 1	// Bit of a safety net; it's impossible to add blood if there's not blood already in the vessel.
+	amt = amt * (src.mob_size/MOB_MEDIUM)
```

Comment:
I think taking out that division from under remove_reagent was all that was needed to fix it in the first place because it was probably causign rounding errors, but rest of it seems fine either way.